### PR TITLE
[MWPW-133606] Aside block automated scripts

### PR DESCRIPTION
**Pull request by chadsunvice**
PR raised in scope of ticket **[[MWPW-133606](https://jira.corp.adobe.com/browse/MWPW-133606)]: Aside Block: Create Nala Automation Test script** .

### CODE REVIEW

**Review by amauch-adobe — CHANGES_REQUESTED** [review]
see my comments above.

File: features/milo/aside.block.spec.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
+module.exports = {
+  name: 'Aside Block',
+  features: [
+    {
+      name: '@Aside-Small',
+      path: '/drafts/nala/blocks/aside/aside-small',
+      browserParams: '?hideGeorouting=on',
+      tags: '@aside, @aside-small, @smoke, @regression @milo',
```
[the remark this answers is not in the document]
Comment (chadsunvice) [reply features/milo/aside.block.spec.js]: Goot catch @amauch-adobe, updated!

File: features/milo/aside.block.spec.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+      name: '@Aside-Split-Medium',
+      path: '/drafts/nala/blocks/aside/aside-split-medium',
+      browserParams: '?hideGeorouting=on',
+      tags: '@aside, @aside-split-medium, @smoke, @regression @milo',
+    },
+    {
+      name: '@Aside-Split-Medium-Half',
+      path: '/drafts/nala/blocks/aside/aside-split-medium-half',
+      browserParams: '?hideGeorouting=on',
+      tags: '@aside, @aside-split-medium, @smoke, @regression @milo',
```
[the remark this answers is not in the document]
Comment (chadsunvice) [reply features/milo/aside.block.spec.js]: Updated!

File: features/milo/aside.block.spec.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
+module.exports = {
+  name: 'Aside Block',
+  features: [
+    {
+      name: '@Aside-Small',
+      path: '/drafts/nala/blocks/aside/aside-small',
+      browserParams: '?hideGeorouting=on',
```
Comment (JackySun9) [inline features/milo/aside.block.spec.js:7]: if browserParams are all the same, why do you put it here?

File: tests/milo/aside.block.test.js [hunk 4]
```diff
@@ -0,0 +1,528 @@
...
+    console.info(`[MiloInfo] Checking page: ${baseURL}${features[0].path}`);
+
+    await test.step('Navigate to page with Aside block', async () => {
+      await page.goto(`${baseURL}${features[0].path}${features[0].browserParams}`);
+      await page.waitForLoadState('domcontentloaded');
+      await expect(page).toHaveURL(`${baseURL}${features[0].path}${features[0].browserParams}`);
+    });
+
+    await test.step('Validate Aside block content', async () => {
+      await expect(Aside.asideSmall).toBeVisible();
```
Comment (JackySun9) [inline tests/milo/aside.block.test.js:18]: Because almost all steps are the same, could you have them in a internal function?

**Review by JackySun9 — CHANGES_REQUESTED** [review]

File: tests/milo/aside.block.test.js (hunk 4 above)
Comment (chadsunvice) [reply tests/milo/aside.block.test.js:18]: The steps are actually not the same. Each Aside block has specific particularities. Maybe some tests have overlapping logic, but the checks are not all the same. Most of them use check different configs of classes and selectors change based on that. 

eg: One Aside block has smallText, others have mediumText. One Aside block has picture, another doesn't. These are particular to every Aside type of block.

**Review by JackySun9 — APPROVED** [review]

**Comment by chadsunvice** [comment]
> see my comments above.

Updated!

File: tests/milo/aside.block.test.js [hunk 5]
```diff
@@ -0,0 +1,528 @@
...
+      // Check Aside block background:
+      const bgdColor = await Aside.asideSplitLargeHalfDark.evaluate((e) => window.getComputedStyle(e).getPropertyValue('background-color'));
+      expect(bgdColor).toBe(Aside.props.background.black);
+    });
+  });
+
+  // Aside Inline Checks:
+  test(`${features[9].name}, ${features[9].tags}`, async ({ page, baseURL }) => {
+    const Aside = new AsideBlock(page);
+    console.info(`[MiloInfo] Checking page: ${baseURL}${features[9].path}`);
```
Comment (skumar09) [inline tests/milo/aside.block.test.js:284]: please change the console.info to format : `console.info(`[Test Page]: ${baseURL}${features[0].path}`);`

**Review by skumar09 — COMMENTED** [review]
@chadsunvice: PR looks good; just following a few points
- do we want to add CSS / attributes validations, Or do you want to take care later?

File: tests/milo/aside.block.test.js [hunk 6]
```diff
@@ -0,0 +1,528 @@
...
+      await expect(Aside.image).toBeVisible();
+      await expect(Aside.actionArea).toBeVisible();
+      await expect(Aside.detailLabel).toBeVisible();
+      await expect(Aside.h3TitleXLarge).toBeVisible();
+      await expect(Aside.textFieldSmall).toBeVisible();
+      // Check Aside block buttons:
+      await expect(Aside.blueButtonCta).not.toBeVisible();
+      await expect(Aside.blackButtonCta).toBeVisible();
+      await expect(Aside.linkTextCta).toBeVisible();
+      expect(await Aside.actionButtons.count()).toEqual(2);
```
Comment (skumar09) [inline tests/milo/aside.block.test.js:61]: - for the Aside block,  actionButtons count will always be 2 or is it configurable? if configuration about adding counts to `.spec.js` or `global`

Comment (chadsunvice) [reply tests/milo/aside.block.test.js:61]: If you look at the entire test-suite, you'll notice there is not real rule enforced throughout the feature. Some Aside blocks have 1 button, some have both, some have only one CTA link. There's really no clear rule here.

File: tests/milo/aside.block.test.js (hunk 5 above)
Comment (chadsunvice) [reply tests/milo/aside.block.test.js:284]: Updated!

**Comment by chadsunvice** [comment]
> @chadsunvice: PR looks good; just following a few points
> 
> * do we want to add CSS / attributes validations, Or do you want to take care later?

I talked to Narcis and also got some feedback from our Devs that there are still Aside changes happening so this will mean most of the checks will become brittle. I would wait for development to stabilise and then take some conclusions on this. 

Also we feel CSS validation isn't something that will change, and attributes are already tested via the selectors that have them in their composition. So I would say wait on this.

**Review by skumar09 — APPROVED** [review]

**Review by amauch-adobe — APPROVED** [review]
Approved.

**Pull request merged**